File: strive/14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(),strs.end());
        string result="";
        for (string str : strs){
            cout<<str<<" ";
        }
        string first = strs[0];
        string last = strs[strs.size()-1];
        for( int i=0 ; i<first.size() ;i++){
            if( first[i] != last[i]){
                return result;
            }
            else{
                result += first[i];
            }   
        }
        return result;
    }
};

File: strive/14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans_str = ""
        strs.sort()

        n = len(strs)
        for i in range( len(strs[0]) ):
            if i==0 and strs[0][i] != strs[n-1][i]:
                return ""
            elif strs[0][i] == strs[n-1][i]:
                ans_str += strs[0][i]
            else:
                break
        return ans_str